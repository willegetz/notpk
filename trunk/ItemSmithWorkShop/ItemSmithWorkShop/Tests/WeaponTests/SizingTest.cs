using Microsoft.VisualStudio.TestTools.UnitTesting;
using AdventureItems;
using ApprovalTests;
using ItemSmithWorkShop.AdventureItems.ExtraordinaryWeaponUtilities;
using MagicWeaponUtilities;

namespace ItemSmithWorkShop.Tests.WeaponTests
{
	[TestClass]
	public class SizingTest
	{
		[TestMethod]
		public void TestSmallSizeWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Small");
			Approvals.Approve(sizedWeapon.DisplaySizedWeapon());
		}

		[TestMethod]
		public void TestLargeSizeWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			Approvals.Approve(sizedWeapon.DisplaySizedWeapon());
		}

		[TestMethod]
		public void TestSmallAdamantineWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Small");
			ExtraordinaryQualityWeapon adamantineWeapon = WeaponItemSmith.OrderSpecialComponent(sizedWeapon, "Adamantine");
			Approvals.Approve(adamantineWeapon.GetItem());
		}

		[TestMethod]
		public void TestLargePlus3Weapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(sizedWeapon, 3);
			Approvals.Approve(magicWeapon.DisplayFullText());
		}
	}
}
