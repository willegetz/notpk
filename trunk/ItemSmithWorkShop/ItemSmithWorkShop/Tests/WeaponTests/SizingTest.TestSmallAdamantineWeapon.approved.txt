Adamantine Dagger [Small]:	'3002 gp'
Weight: '0.665 pound(s)'
To Hit: '+1'
Damage: '1d3'
Hardness: '5'
Hit Points: '1'
	Adamantine weapons are always considered to be masterwork quality.
	Adamantine bypasses a damage reduction of 20.
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.